zcash_client_backend/src/welding_rig.rs

```diff
@@ -141,16 +144,65 @@ impl ScanningKey for SaplingIvk {
 /// [`IncrementalWitness`]: zcash_primitives::merkle_tree::IncrementalWitness
 /// [`WalletShieldedOutput`]: crate::wallet::WalletShieldedOutput
 /// [`WalletTx`]: crate::wallet::WalletTx
-pub fn scan_block<P: consensus::Parameters, K: ScanningKey>(
+pub fn scan_block<P: consensus::Parameters + Send + 'static, K: ScanningKey>(
+    params: &P,
+    block: CompactBlock,
+    vks: &[(&AccountId, &K)],
+    nullifiers: &[(AccountId, Nullifier)],
+    tree: &mut CommitmentTree<Node>,
+    existing_witnesses: &mut [&mut IncrementalWitness<Node>],
+) -> Vec<WalletTx<K::Nf>> {
+    scan_block_with_runner(
+        params,
+        block,
+        vks,
+        nullifiers,
+        tree,
+        existing_witnesses,
+        None,
+    )
+}
+
+pub(crate) fn add_block_to_runner<P: consensus::Parameters + Send + 'static>(
+    params: &P,
+    block: CompactBlock,
+    batch_runner: &mut BatchRunner<SaplingDomain<P>, CompactOutputDescription>,
```

Comment:
How hard will it be to adapt this to cover Orchard batch scanning? I'm concerned that we could end up duplicating quite a lot of code and I want to minimise that as far as possible.

Reply:
I've already started the process of adding Orchard to the zcashd version of this in https://github.com/zcash/zcash/pull/6105. Once that is done I'll apply the same changes to this code, assuming it still exists (and hasn't been replaced with async logic).